### ISSUE TITLE
torch.cos has high variance required for vllm RoPE implementation

### ISSUE THREAD

**Issue by dilipgb** [torch-spyre/torch-spyre#1668]
### What happened?

We need to implement RoPE embeddings in vllm project. We are trying to use cos/sin functions we are getting tokens different than expected tokens. We have tried using torch.float32 as well torch.float16 for computing cos/sin cache but results are not as expected. torch.flaot16 results in garbage tokens (\n\n\n\n). To confirm numerical correctness, ran comparision between CPU and spyre for cos/sin and results had high variance in cos computation.

### What did you expect to happen?

We need numerical variance should be minimal upto e-5 and above to correctly process logits in vllm.

### How can we reproduce it?

Here is the sample script used and generated output for both torch.float32 torch.float16 dtypes.
```
(dt-inductor) [dilip@dilip-spyre-dev-pf ~]$ cat test_cos_sin_var.py 
import torch

# deterministic
torch.manual_seed(0)

seq_len = 1024
head_dim = 128

#RoPE frequencies input
theta = torch.randn(seq_len, head_dim, dtype=torch.float16)

cos_cpu = torch.cos(theta)
sin_cpu = torch.sin(theta)

theta_spyre = theta.to("spyre")  

cos_spyre = torch.cos(theta_spyre)
sin_spyre = torch.sin(theta_spyre)

print("theta dtype:", theta.dtype)
print("theta_spyre dtype:", theta_spyre.dtype)

diff = (theta - theta_spyre.cpu()).abs()
print("theta max diff:", diff.max().item())

# bring back for comparison
cos_spyre = cos_spyre.cpu()
sin_spyre = sin_spyre.cpu()

print(theta.is_leaf, theta_spyre.is_leaf)

def compare(name, ref, test):
    abs_err = (ref - test).abs()
    rel_err = abs_err / (ref.abs() + 1e-8)

    print(f"\n{name}:")
    print("  max abs error:", abs_err.max().item())
    print("  mean abs error:", abs_err.mean().item())
    print("  max rel error:", rel_err.max().item())
    print("  mean rel error:", rel_err.mean().item())

compare("cos", cos_cpu, cos_spyre)
compare("sin", sin_cpu, sin_spyre)
```

### Any environmental details we need to know?

NA

### Anything else we need to know?

releavant PR in sendnn-infernce repo. https://github.com/torch-spyre/sendnn-inference/pull/881

### Relevant log output

```Shell
FP16:


(dt-inductor) [dilip@dilip-spyre-dev-pf ~]$ python test_cos_sin_var.py 
/home/dilip/test_cos_sin_var.py:17: FallbackWarning: aten.cos.out is falling back to cpu
  cos_spyre = torch.cos(theta_spyre)
/home/dilip/test_cos_sin_var.py:18: FallbackWarning: aten.sin.out is falling back to cpu
  sin_spyre = torch.sin(theta_spyre)
theta dtype: torch.float16
theta_spyre dtype: torch.float16
theta max diff: 0.001953125
True True

cos:
  max abs error: 0.00244140625
  mean abs error: 0.0003325939178466797
  max rel error: 1.9814453125
  mean rel error: 0.002254486083984375

sin:
  max abs error: 0.00244140625
  mean abs error: 0.000316619873046875
  max rel error: nan
  mean rel error: nan
(dt-inductor) [dilip@dilip-spyre-dev-pf ~]$ python test_cos_sin_var.py 
/home/dilip/test_cos_sin_var.py:17: FallbackWarning: aten.cos.out is falling back to cpu
  cos_spyre = torch.cos(theta_spyre)
/home/dilip/test_cos_sin_var.py:18: FallbackWarning: aten.sin.out is falling back to cpu
  sin_spyre = torch.sin(theta_spyre)
theta dtype: torch.float16
theta_spyre dtype: torch.float16
theta max diff: 0.001953125
True True

cos:
  max abs error: 0.00244140625
  mean abs error: 0.0003325939178466797
  max rel error: 1.9814453125
  mean rel error: 0.002254486083984375

sin:
  max abs error: 0.00244140625
  mean abs error: 0.000316619873046875
  max rel error: nan
  mean rel error: nan


FP32

(dt-inductor) [dilip@dilip-spyre-dev-pf ~]$ python test_cos_sin_var.py 
/home/dilip/test_cos_sin_var.py:17: FallbackWarning: aten.cos.out is falling back to cpu
  cos_spyre = torch.cos(theta_spyre)
/home/dilip/test_cos_sin_var.py:18: FallbackWarning: aten.sin.out is falling back to cpu
  sin_spyre = torch.sin(theta_spyre)
theta dtype: torch.float32
theta_spyre dtype: torch.float32
theta max diff: 0.0

cos:
  max abs error: 0.00015109777450561523
  mean abs error: 7.318805728573352e-06
  max rel error: 0.00015288268332369626
  mean rel error: 1.0222249329672195e-05

sin:
  max abs error: 0.0
  mean abs error: 0.0
  max rel error: 0.0
  mean rel error: 0.0
```

**Comment by moriohara** [comment]
@dilipgb does this help? https://github.com/torch-spyre/torch-spyre/pull/4287
